Mathlib/GroupTheory/GroupAction/Hom.lean

```diff
@@ -92,18 +113,35 @@ class MulActionSemiHomClass (F : Type*)
 
 export MulActionSemiHomClass (map_smulₛₗ)
 
+/-- `AddActionSemiHomClass F φ X Y` states that
+  `F` is a type of morphisms which are `φ`-equivariant.
+
+You should extend this class when you extend `AddActionHom`. -/
+class AddActionSemiHomClass (F : Type*)
+    {M N : outParam Type*} (φ : outParam (M → N))
+    (X Y : outParam Type*) [VAdd M X] [VAdd N Y] [FunLike F X Y] : Prop where
+  /-- The proposition that the function preserves the action. -/
+  map_vaddₛₗ : ∀ (f : F) (c : M) (x : X), f (c +ᵥ x) = (φ c) +ᵥ (f x)
+
+attribute [to_additive] MulActionSemiHomClass
```

Comment:
Can you define this before `MulActionSemiHomClass` so that you can directly do
```
@[to_additive]
class MulActionSemiHomClass
...
```

Reply:
I am not sure I understand what you suggest.

Reply:
I tried something and it seems to work.